inst/private/assert_have_python_and_sympy.m

```diff
@@ -146,6 +146,67 @@ function assert_have_python_and_sympy (pyexec, verbose)
   end
 
   if (verbose)
+    disp ('')
+    disp ('Good, a working version of SymPy is installed.')
+    disp ('')
+    disp ('')
+    disp ('Python XML Parsing and DOM Support')
+    disp ('----------------------------------')
+    disp ('')
+    disp ('The XML DOM library is used by Symbolic for passing values to and from Python.')
+    disp ('Some older versions of Python formatted XML output differently.  As long as you')
+    disp ('have any reasonably recent version of Python, this should pass.')
+    disp ('')
+  end
+
+  script = 'import xml.dom.minidom as minidom;';
+  script = [script ' doc = minidom.parseString(\"<item>value</item>\");'];
+  script = [script ' print(doc.toprettyxml(indent=\"\"))'];
+
+  if (verbose)
+    fprintf ('Attempting to run %s -c "%s"\n\n', pyexec, script);
+  end
+
+  [status, output] = system([pyexec ' -c "' script '"']);
+  if (verbose)
+    status
+    output
+  end
+
+  if (status ~= 0)
+    if (~ verbose)
+      error ('OctSymPy:noxmldom', ...
+             ['Python cannot import XML DOM: is this an unusual Python setup?\n' ...
+              '    Try "sympref diagnose" for more information.'])
+    end
+    disp ('')
+    disp ('Unfortunately status was non-zero: probably Python cannot import xml.dom.')
+    disp ('')
+    disp ('  * Is there an error message above?')
+    disp ('')
+    disp ('  * Are you using an unusual Python installation?  If not, please try to')
+    disp ('    reinstall it and try again.')
+    return
+  end
+
+  xmlout = strtrim (output);
+
+  if (isempty (strfind (xmlout, '<item>value</item>')))
+    if (~ verbose)
+      error ('OctSymPy:oldxmldom', ...
+             ['Python XML output is not compatible, probably an old version of Python?\n' ...
+              '    Try "sympref diagnose" for more information.'])
+    end
+    disp ('**** Your Python is too old! ****')
+    disp ('')
+    disp ('The XML output shown above must have "<item>value</item>" on the same line.')
+    disp ('If your Python interpreter is an old version of Python 2, or Python 3 older')
+    disp ('than 3.2.3, please try upgrading to a more recent Python to use the Symbolic')
+    disp ('package.')
```

Comment:
Maybe its just hard to phrase this briefly but looks a little odd to explicitly mention py3 version but not not py2 version.

Reply:
Yeah this is the awkward part to write, especially for people who don't care what the difference is between Python 2 and 3. Maybe just simplify it like

> If your Python interpreter is an older release, please try upgrading to a more recent Python to use the Symbolic package.  Symbolic works with any Python 2.7 version 2.7.3 or newer and any Python 3 version 3.2.3 or newer.

Reply:
#bikeshed ;-)

> If your Python interpreter is an older release, please try upgrading to a
> more recent version.  Symbolic should work with Python 2 version 2.7.3
> or newer and Python 3 version 3.2.3 or newer.

or

> If your Python interpreter is an older release, please try upgrading to a
> more recent version.  Symbolic should work with Python 2 (version 2.7.3
> or newer) and Python 3 (version 3.2.3 or newer).